x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/cache/SearchableSnapshotsPersistentCacheIntegTests.java

```diff
@@ -143,7 +91,9 @@ public void testCacheDirectoriesRemovedOnStartup() throws Exception {
             .getIndex();
 
         final IndexService indexService = internalCluster().getInstance(IndicesService.class, dataNode).indexService(restoredIndex);
-        final Path shardCachePath = CacheService.getShardCachePath(indexService.getShard(0).shardPath());
+        final ShardPath shardPath = indexService.getShard(0).shardPath();
+        final Path shardCachePath = CacheService.getShardCachePath(shardPath);
+
         assertTrue(Files.isDirectory(shardCachePath));
         final Set<Path> cacheFiles = new HashSet<>();
         try (DirectoryStream<Path> snapshotCacheStream = Files.newDirectoryStream(shardCachePath)) {
```

Comment:
It runs concurrently with prewarming but I think it is OK. I expect Lucene to read at least the header checksum of every file when opening the Directory and that should constitute the set of cache files which should remain unchanged (because the cache size in this test is set to 1GB so we should not expect any cache eviction too).

Reply:
:+1: